packages/core/src/root.ts

```diff
@@ -127,9 +132,15 @@ export function createRemoteRoot<
         children.set(component, strict ? Object.freeze([]) : []);
       }
 
+      componentsSet.add(component);
+
       return (component as unknown) as RemoteComponent<typeof type, Root>;
     },
     createText(content = '') {
+      if (!validateAllowedComponent('Text' as any, components)) {
```

Comment:
This still gets messy if you have a component called `Text`, which we do at least in checkout's Argo API. I don't think we should ever restrict text nodes.

Reply:
Fixed!